assets/js/session/index.js

```diff
@@ -522,6 +529,18 @@ function saveNotebook(hook) {
   hook.pushEvent("save", {});
 }
 
+function captureCodeSnippet(hook) {
+  if (hook.state.focusedCellId) {
+    html2canvas(
+      document.querySelector("#cell-" + hook.state.focusedCellId)
+    ).then((canvas) => {
+      const dataURL = canvas.toDataURL("image/png");
+      const newTab = window.open("about:blank", "livebook code snippet");
+      newTab.document.write("<img src='" + dataURL + "' alt='code-snippet' />");
```

Comment:
To avoid the "continuously loading" effect you need to call `document.close()` after writing everything. Also for generating the image html we could rely on the DOM api like, so something like this:

```js
const image = new Image();
image.src = dataURL;
image.alt = "livebook code snippet";
newTab.document.write(image.outerHTML);
newTab.document.close();
newTab.document.title = "Livebook code snippet";
```

Reply:
Fixed!
Thanks much